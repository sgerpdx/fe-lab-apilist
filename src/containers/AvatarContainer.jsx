import React, { Component } from 'react';
import { fetchCharacters } from '../services/AvatarAPI';
import CharactersList from '../components/app/characters/CharactersList';

export default class AvatarContainer extends Component {
  state = {
    thingOne: '',
    loading: true,
    characterData: [],
  };

  componentDidMount = async () => {
    const newData = await fetchCharacters();
    this.setState({ characterData: newData, loading: false });
  };

  render() {
    if (this.state.loading) return <h3>Loading...</h3>;

    return (
      <>
        <h2>Container</h2>
        <CharactersList characters={this.state.characterData} />
      </>
    );
  }
}
