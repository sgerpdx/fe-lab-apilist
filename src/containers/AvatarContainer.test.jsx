import React from 'react';
import { render, screen } from '@testing-library/react';
import AvatarContainer from './AvatarContainer';
//import userEvent from '@testing-library/user-event';
//import { rest } from 'msw';
//import { setupServer } from 'msw/node';

describe('list container component for Avatar characters', () => {
  it('displays the correct stuff', async () => {
    render(<AvatarContainer />);

    const pageLoading = await screen.getByText('Loading...');
    expect(pageLoading).toMatchSnapshot();
  });
});
