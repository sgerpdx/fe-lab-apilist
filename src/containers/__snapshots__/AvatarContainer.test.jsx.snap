// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list container component for Avatar characters displays a list of characters 1`] = `
<h3>
  Loading...
</h3>
`;
